test/crd.go

```diff
@@ -35,6 +35,7 @@ type ResourceNames struct {
 	TrafficTarget string
 	URL           *url.URL
 	Image         string
+	BYOName       string
```

Comment:
Rather than passing this everywhere, we can just pass a new option to write the name?  We probably even have a function to set this.